File: result/metagpt/claude/humaneval/66.c
#include <stdio.h>
#include <string.h>

int digitSum(const char* s) {
    int sum = 0;
    for (int i = 0; i < strlen(s); i++) {
        if (s[i] >= 65 && s[i] <= 90) {
            sum += s[i];
        }
    }
    return sum;
}

File: result/metagpt/deepseek/humaneval/66.rs
pub fn digit_sum(s: &str) -> u32 {
    s.chars()
        .filter(|c| c.is_ascii_uppercase())
        .map(|c| c as u32)
        .sum()
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_digit_sum() {
        assert_eq!(digit_sum("ABC"), 198); // A(65) + B(66) + C(67) = 198
        assert_eq!(digit_sum("aBc"), 66);
        assert_eq!(digit_sum(""), 0);
    }
}